crates/turborepo-lib/src/commands/link.rs

```diff
@@ -404,14 +405,19 @@ fn add_turbo_to_gitignore(base: &CommandBase) -> Result<()> {
         writeln!(gitignore, ".turbo")?;
     } else {
         let gitignore = File::open(&gitignore_path)?;
-        let mut lines = io::BufReader::new(gitignore).lines();
-        let has_turbo = lines.any(|line| line.map_or(false, |line| line.trim() == ".turbo"));
+        let mut lines = io::BufReader::new(&gitignore).lines();
+        let has_turbo = lines
+            .find(|line| matches!(line, Ok(line_value) if turbo_test_regex.is_match(line_value)))
+            .is_some();
+
         if !has_turbo {
             let mut gitignore = OpenOptions::new()
                 .read(true)
                 .append(true)
                 .open(&gitignore_path)?;
-            writeln!(gitignore, ".turbo")?;
+
+            // Defensively ensure that it has a preceding newline.
+            writeln!(gitignore, "\n.turbo")?;
```

Comment:
Always precede with a newline. No need to be clever and check for existing newlines.

Reply:
Also, if somebody complains about a bonus newline, my "fix" to this is:

```
writeln!(gitignore, "\n# Ignore the Turborepo configuration directory.\n.turbo")?;
```